mono/metadata/threads-types.h

```diff
@@ -83,7 +89,7 @@ gboolean ves_icall_System_Threading_Thread_Join_internal(MonoThread *this_obj, i
 gint32 ves_icall_System_Threading_Thread_GetDomainID (void);
 gboolean ves_icall_System_Threading_Thread_Yield (void);
 MonoString* ves_icall_System_Threading_Thread_GetName_internal (MonoInternalThread *this_obj);
-void ves_icall_System_Threading_Thread_SetName_internal (MonoInternalThread *this_obj, MonoString *name);
+void ves_icall_System_Threading_Thread_SetName_internal (MonoInternalThreadHandle thread, const char* name_utf8, int length, MonoError *error);
```

Comment:
We use `gint32` instead of `int` for data coming from managed for guaranteed size.

Reply:
Sure. Given that gtypes not usable publicly, stop using them?

Reply:
Understood but why bother using them at all? Vs int32_t? They are slightly shorter, granted. They might have more consistency between int & long on 32bit also.